draft-ietf-quic-tls.md

```diff
@@ -703,6 +704,22 @@ ensuring that the keys are different for each version of QUIC. This prevents a
 middlebox that only recognizes one version of QUIC from seeing or modifying the
 contents of handshake packets from future versions.
 
+Note(1):
+
+: The Destination Connection ID is of arbitrary length, and it could be zero
+  length if the server sends a Retry packet with a zero-length Source Connection
+  ID field.  In this case, the handshake keys provide no assurance to the client
+  that the server received its packet; the client has to rely on the exchange
+  that included the Retry packet for that property.
+
+Note(2):
+
+: HKDF-Extract uses HMAC {{?RFC2104}}, which produces the same output for values
+  ending in zero octets and values that omit those octets.  Thus,
+  0x12345678900000 and 0x1234567890 produce the same result.  A client is
+  required to randomize the value of Destination Connection ID, which avoids
+  this problem.
+
 
```

Comment:
Can we avoid requiring it to be random, or allow it to be only partially random? If a client has intimate knowledge of the server infrastructure, it might want to route the packet, or add some guidance in the initial destination ID.

Reply:
Tempting to simply require that the final octet not be zero.  If the client has knowledge of how to generate a Connection ID, it can respin in the 1/16 chance it generates one that ends in zeros.

Or alternatively, append a fixed non-zero octet to the Connection ID to side-step this problem.

Reply:
@MikeBishop: as noted above, I don't think this is necessary.